test/syntax.jl

```diff
@@ -317,19 +317,16 @@ let p = 15
     @test 2p+1 == 31  # not a hex float literal
 end
 
-function test_parseerror(str, msg)
-    try
-        Meta.parse(str)
-        @test false
-    catch e
-        @test isa(e,ParseError) && e.msg == msg
-    end
+macro test_parseerror(str, msg)
+    ex = :(@test_throws ParseError($(esc(msg))) Meta.parse($(esc(str))))
+    ex.args[2] = __source__
```

Comment:
@JeffBezanson this is an example of why we needed https://github.com/JuliaLang/julia/pull/44995, since the `ex` object here fails to track the top-level line number after all of the code expansion